docs/pages/inboxes/disappearing-messages.md

```diff
@@ -0,0 +1,84 @@
+# Support disappearing messages with XMTP
+
+Disappearing messages are messages that are intended to be visible to users for only a short period of time. After the message expiration time passes, the messages are removed from the UI and deleted from local storage so the messages are no longer accessible to conversation participants.
+
+Disappearing message behavior is enforced by apps, meaning that apps are responsible for removing messages from their UIs and local storage based on conditions set at the conversation level. Conversation participants using apps that support disappearing messages will have a UX that honors the message expiration conditions. 
+
+However, it's important to understand that:
+
+- A conversation participant using an app that doesn't support disappearing messages won't experience disappearing message behavior.
+- Messages aren't deleted from the XMTP network.
+
+:::tip
+
+Therefore, disappearing messages should be understood as best-effort, app-level privacy that helps avoid leaving an easily accessible record in a messaging UI. However, it's not a guaranteed, system-wide “message self-destruct.”
+
+:::
+
+## Implement disappearing messages
+
+### Enable disappearing messages for a conversation
+
+When creating or updating a conversation, admins can set disappearing message expiration conditions.
+
+This includes setting the following conditions expressed in nanoseconds (ns):
+
+- `from_ns`: Starting timestamp from which the message lifespan is calculated
+- `in_ns`: Duration of time during which the message should remain visible to conversation participants
+
+For example:
+
+1. Set `from_ns` to the current time, such as `1738620126404999936` (nanoseconds since the Unix epoch of January 1, 1970).
+2. Set `in_ns` to the message lifespan, such as 1800000000000000 (30 minutes).
+3. Use `from_ns` and `in_ns` to calculate the message expiration time of `1738620126404999936 + 1800000000000000 = 1740420126404999936`.
+
+To learn more see [conversation.rs](https://github.com/xmtp/libxmtp/blob/main/bindings_node/src/conversation.rs#L49).
+
+### Send a disappearing message
+
+When sending a message, it abides by message expiration conditions set for the conversation. For example:
+
+:::code-group
+
+```tsx [Browser]
+// add code sample
+```
+
+```tsx [Node]
+// add code sample
+```
+
+```tsx [React Native]
+// add code sample
+```
+
+```kotlin [Kotlin]
+// add code sample
+```
+
+```swift [Swift]
+// add code sample
+```
+
+:::
+
+### Automatic deletion from local storage
+
+A background worker runs every one second to clean up expired disappearing messages. The worker automatically deletes expired messages from local storage. No additional action is required by integrators.
+
+To learn more about the background worker, see [link to background worker code](#todo).
```

Comment:
@mchenani thank you for the suggestion - yes, it would be great to link to relevant background worker code - could you suggest a link, please?

Reply:
https://github.com/xmtp/libxmtp/blob/main/xmtp_mls/src/groups/disappearing_messages.rs#L68
This is the worker code, also you can link the whole file, since it only contains the disappearing messages cleaner worker.